src/mlpack/core/optimizers/sgd/sgd.hpp

```diff
@@ -96,7 +97,8 @@ class SGD
       const double stepSize = 0.01,
       const size_t maxIterations = 100000,
       const double tolerance = 1e-5,
-      const bool shuffle = true);
+      const bool shuffle = true,
+      const UpdatePolicy updatePolicy = VanillaUpdate());
```

Comment:
I think if you like to use MomentumSGD without passing the Momentum policy this fails, because  in that case UpdatePolicy is MomentumUpdate and can't be assigned with VanillaUpdate(), instead if you use ``const UpdatePolicy updatePolicy = UpdatePolicy()`` it should be fine.